### PR TITLE
[Fix #10529] Fix autocorrect for `Style/SoleNestedConditional`

### LINKED ISSUE #10529: rubocop -a cause logic error with Style/SoleNestedConditional cop on version 1.27 + ruby 3.1
Following is a sample.

```rb
x = 100
y = true
z = true
foo = true

if x.instance_of? Integer
  if !y && !z
    return puts 'ok' if foo
  end
end
```

After run rubocop -a or used with solargraph, it will be change to:

```rb
x = 100
y = true
z = true
foo = true

return puts 'ok' if x.instance_of? Integer && (!y && !z) && foo
```

That was not expected, it should be:

```rb
x = 100
y = true
z = true
foo = true

return puts 'ok' if (x.instance_of? Integer) && (!y && !z) && foo
```

### PR DESCRIPTION
**Pull request by nobuyo**
Fixes #10529

-----------------

Before submitting the PR make sure the following are checked:

* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by nobuyo** [comment]
CI / windows 3.1 has failed, but I have a feeling it's not due to my changes 🤔

**Comment by ydah** [comment]
I don't know if this is the right way to handle this, but wouldn't running CI again solve the problem?
I sometimes encounter that too🤔

**Comment by koic** [comment]
Yep, probably a false alarm due to an unstable test.

File: changelog/fix_autocorrect_for_style_sole_nested_confitional.md [hunk 1]
```diff
@@ -0,0 +1 @@
+* [#10529](https://github.com/rubocop/rubocop/issues/10529): Fix autocorrect for `Style/SoleNestedConditional` causes logical error when using a outer condition of send_type by omitting parentheses for method arguments. ([@nobuyo][])
```
Comment (koic) [inline changelog/fix_autocorrect_for_style_sole_nested_confitional.md:1]: Can you replace "send_type" with "method call"? 

```suggestion
* [#10529](https://github.com/rubocop/rubocop/issues/10529): Fix autocorrect for `Style/SoleNestedConditional` causes logical error when using a outer condition of method call by omitting parentheses for method arguments. ([@nobuyo][])
```

File: spec/rubocop/cop/style/sole_nested_conditional_spec.rb [hunk 2]
```diff
@@ -348,6 +348,24 @@ def foo
     RUBY
   end
 
+  it 'registers an offense and corrects for multiple nested conditionals with using send_type outer condition by omitting parentheses' do
```
Comment (koic) [inline spec/rubocop/cop/style/sole_nested_conditional_spec.rb:351]: Ditto.

```suggestion
  it 'registers an offense and corrects for multiple nested conditionals with using method call outer condition by omitting parentheses' do
```

**Comment by nobuyo** [comment]
I'd like to do `git rebase -i master` again.

**Comment by koic** [comment]
This looks good. Can you squash your commits into one?

**Comment by nobuyo** [comment]
Thanks for your review, updated branch.

**Comment by koic** [comment]
Thanks!

**Comment by zw963** [comment]
This fix works on 1.28.2, thank you.

**Pull request merged**